src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -0,0 +1,324 @@
+                _compilationState = compilationState;
+                _diagnostics = diagnostics;
+                _debugDocumentProvider = debugDocumentProvider;
+                _methodHasExplicitBlock = MethodHasExplicitBlock(method);
+                _factory = factory;
+            }
+        }
+
+        public BoundStatement AddInstrumentationPrologue(BoundBlock rewrittenMethodBody, ref ImmutableArray<SourceSpan> dynamicAnalysisSpans)
+        {
+            if (_doInstrumentation)
+            {
+                dynamicAnalysisSpans = _spansBuilder.ToImmutableAndFree();
+                // PROTOTYPE (https://github.com/dotnet/roslyn/issues/10411): In the future there will be multiple analysis kinds.
+                const int analysisKind = 0;
+
+                ArrayTypeSymbol modulePayloadType = ArrayTypeSymbol.CreateCSharpArray(_compilationState.Compilation.Assembly, _payloadType);
+
+                // Synthesize the initialization of the instrumentation payload array, using concurrency-safe code:
+                //
+                // var payload = PID.PayloadRootField[methodIndex];
+                // if (payload == null)
+                //     payload = Instrumentation.CreatePayload(mvid, methodIndex, ref PID.PayloadRootField[methodIndex], payloadLength);
+
+                BoundStatement payloadInitialization = _factory.Assignment(_factory.Local(_methodPayload), _factory.ArrayAccess(_factory.InstrumentationPayloadRoot(analysisKind, modulePayloadType), ImmutableArray.Create(_factory.MethodDefIndex(_method))));
+                BoundExpression mvid = _factory.ModuleVersionId();
+                BoundExpression methodToken = _factory.MethodDefIndex(_method);
+                BoundExpression payloadSlot = _factory.ArrayAccess(_factory.InstrumentationPayloadRoot(analysisKind, modulePayloadType), ImmutableArray.Create(_factory.MethodDefIndex(_method)));
+                BoundStatement createPayloadCall = _factory.Assignment(_factory.Local(_methodPayload), _factory.Call(null, _createPayload, mvid, methodToken, payloadSlot, _factory.Literal(dynamicAnalysisSpans.Length)));
+
+                BoundExpression payloadNullTest = _factory.Binary(BinaryOperatorKind.ObjectEqual, _factory.SpecialType(SpecialType.System_Boolean), _factory.Local(_methodPayload), _factory.Null(_payloadType));
+                BoundStatement payloadIf = _factory.If(payloadNullTest, createPayloadCall);
+
+                // There must be a sequence point before any executable code. A sequence point
+                // won't be generated automatically for the synthesized prologue because the prologue is compiler generated, so
+                // force the generation of a sequence point for the prologue.
+                payloadInitialization = _factory.SequencePoint(payloadInitialization.Syntax, payloadInitialization);
```

Comment:
This is a suspect. This instrumenter should not be dealing with sequence points, if possible.